src/irr/asset/COBJMeshFileLoader.cpp

```diff
@@ -355,490 +734,377 @@ asset::SAssetBundle COBJMeshFileLoader::loadAsset(io::IReadFile* _file, const as
+            //position
+            vtxParams.attributes[POSITION].binding = BND_NUM;
+            vtxParams.attributes[POSITION].format = EF_R32G32B32_SFLOAT;
+            vtxParams.attributes[POSITION].relativeOffset = offsetof(SObjVertex, pos);
+            //normal
+            vtxParams.attributes[NORMAL].binding = BND_NUM;
+            vtxParams.attributes[NORMAL].format = EF_A2B10G10R10_SNORM_PACK32;
+            vtxParams.attributes[NORMAL].relativeOffset = offsetof(SObjVertex, normal32bit);
+            //uv
+            if (hasUV)
             {
-                core::vectorSIMDf v1,v2,v3;
-                v1.set(ctx.Materials[m]->Vertices[ctx.Materials[m]->Indices[i+0]].pos);
-                v2.set(ctx.Materials[m]->Vertices[ctx.Materials[m]->Indices[i+1]].pos);
-                v3.set(ctx.Materials[m]->Vertices[ctx.Materials[m]->Indices[i+2]].pos);
-                v1.makeSafe3D();
-                v2.makeSafe3D();
-                v3.makeSafe3D();
-                core::vectorSIMDf normal(core::plane3dSIMDf(v1, v2, v3).getNormal());
-                newNormals[ctx.Materials[m]->Indices[i+0]] += normal;
-                newNormals[ctx.Materials[m]->Indices[i+1]] += normal;
-                newNormals[ctx.Materials[m]->Indices[i+2]] += normal;
+                vtxParams.attributes[UV].binding = BND_NUM;
+                vtxParams.attributes[UV].format = EF_R32G32_SFLOAT;
+                vtxParams.attributes[UV].relativeOffset = offsetof(SObjVertex, uv);
             }
-            for (size_t i=0; i<ctx.Materials[m]->Vertices.size(); i++)
+
+            ICPURenderpassIndependentPipeline* pipeline = submeshes[i]->getPipeline();
+            if (!pipeline)
+                continue;
+            const auto& mtl = static_cast<const CMTLPipelineMetadata*>(pipeline->getMetadata())->getMaterial();
+            auto shaders = getShaders(hasUV, mtl);
+
+            pipeline->getRasterizationParams().faceCullingMode = EFCM_BACK_BIT;
+            pipeline->getVertexInputParams() = vtxParams;
+            pipeline->setShaderAtIndex(0u, shaders.first.get());
+            pipeline->setShaderAtIndex(4u, shaders.second.get());
+            if (hasUV && mtl.maps[CMTLPipelineMetadata::SMtl::EMP_OPACITY].size())
             {
-                ctx.Materials[m]->Vertices[i].normal32bit = asset::quantizeNormal2_10_10_10(newNormals[i]);
+                auto& blendParams = pipeline->getBlendParams();
+                for (uint32_t i = 0u; i < SBlendParams::MAX_COLOR_ATTACHMENT_COUNT; ++i)
+                {
+                    blendParams.blendParams[i].blendEnable = true;
+                    blendParams.blendParams[i].srcColorFactor = EBF_SRC_ALPHA;
+                    blendParams.blendParams[i].srcAlphaFactor = EBF_SRC_ALPHA;
+                    blendParams.blendParams[i].dstColorFactor = EBF_ONE_MINUS_SRC_ALPHA;
+                    blendParams.blendParams[i].dstAlphaFactor = EBF_ONE_MINUS_SRC_ALPHA;
+                }
             }
```

Comment:
in the MTL, don't need to check for UVs, but there's not only the opacity map, there's also the transmission map that you'd need to check for (different blend equation needed though, probably dual source blending or sth its like having a separate alpha for every RGB channel)